### PR TITLE
Convert build process to CMake + make/nmake to add built in tests and…

### PR DESCRIPTION
**Pull request by ScottBailey**
- Added outputs to build a python free package (`ninja-base`) and replaced the existing one with a virtual one (`ninja`) that depends on `ninja-base` and `python`.
- Switched from the legacy `configure.py --bootstrap` build to the new CMake builds. This allows us to run ninja's tests and install it in the standard way.

### CODE REVIEW

**Review by skupr-anaconda — APPROVED** [review]

**Review by varlackc — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,23 +11,30 @@ source:
   sha256: ce35865411f0490368a8fc383f29071de6690cbadc27704734978221f25e2bed
 
 build:
-  number: 3
   skip: true  # [win and vc!=14]
```
Comment (cbouss) [inline recipe/meta.yaml]: I think the restriction on VS2015 only can be removed.
There used to be a run requirement introduced here:
https://github.com/AnacondaRecipes/ninja-feedstock/commit/5194af92468b8baf804da715ecc7830df6972f8d
But this was removed there:
https://github.com/AnacondaRecipes/ninja-feedstock/commit/e0489d9cb6e68e996dad532e0b7694ea6a012d9b

**Review by cbouss — CHANGES_REQUESTED** [review]
Looks good, but I would remove the VS2015 requirement for windows.

**Review by skupr-anaconda — APPROVED** [review]

**Review by cbouss — APPROVED** [review]
looks good!

**Pull request merged**